### PR TITLE
EventScoreAPI

### PR DESCRIPTION
**Pull request by michmitz**
**What this does:**
* Creates a class, EventScoreAPI
* EventScoreAPI has two properties: score and eventArr
* Method - calcHighestScoreArrSequence - returns a contigous sub array derived from eventArr property, no greater than five elements, with the highest score. Score is determined by event type on event element:
  - screenshot = 3
  - views = 2
  - new message = 1

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -223,4 +223,116 @@ describe('Tests will go here!', () => {
       } 
   })
 })
- 
\ No newline at end of file
+
+describe('EventScore API', () => {
+    it('Returns an entire array if it is five events or less', () => {
```
Comment (ianandrewson) [inline src/index.test.ts:228]: Might be worth having tests for edge cases: arrays of length 0 and 5.

File: src/index.ts [hunk 2]
```diff
@@ -82,3 +82,56 @@ export class UserAPI {
...
+            this.eventArr.map((event, i) => {
+                const { eventType } = event
+                if (eventType === "screenshot") {
+                    tempCount += 3
+                    } else if (eventType === "view") {
+                        tempCount += 2
+                    } else tempCount += 1
+            
+                if (i > 4) {
+                    if (eventType[i - 1] === "screenshot") {
```
Comment (ianandrewson) [inline src/index.ts]: Are you sure you want to subtract from tempCount the most recent value added?

File: src/index.ts [hunk 3]
```diff
@@ -82,3 +82,56 @@ export class UserAPI {
...
+        let tempCount: number = 0
+        let highestScoreArr: ReadonlyArray<Event> = []
+
+        if (this.eventArr.length <= 5) {
+            highestScoreArr = this.eventArr
+            return highestScoreArr
+        } else {
+            this.eventArr.map((event, i) => {
+                const { eventType } = event
+                if (eventType === "screenshot") {
```
Comment (ianandrewson) [inline src/index.ts:114]: Yup, can definitely refactor the conversion of eventType to a scoring value.

File: src/index.ts [hunk 4]
```diff
@@ -82,3 +82,56 @@ export class UserAPI {
...
+    private eventArr: ReadonlyArray<Event>
+
+    constructor(eventArr: ReadonlyArray<Event>) {
+        this.eventArr = eventArr
+        this.score = 0
+    }
+
+    public calcHighestScoreArrSequence = (): ReadonlyArray<Event> => {
+        let tempCount: number = 0
+        let highestScoreArr: ReadonlyArray<Event> = []
```
Comment (ianandrewson) [inline src/index.ts:106]: While insignificant in this case, as highestScoreArr is going to have max length of 5, it is possible to solve this without storing the returned highestScoreArr, and instead just directly return it.

**Review by ianandrewson — COMMENTED** [review]
Definitely on the right track with your algorithm.
